### ISSUE TITLE
Render thread slowed down by communication thread

### ISSUE THREAD

**Issue by skywolf829** [skywolf829/GSTK#27]
The GIL may be impacting our rendering speed performance - when the communication thread is disabled, the backend can train/render much faster - up to 3x it seems. There may be a better way to engineer the backend threads to be more efficient.

One solution might be to use multiprocessing to launch the communicator as its own process, so it can truly send and receive messages in parallel to the rendering/training threads. Then, we just need some way for the 2 processes to talk to each other without introducing another thread and overhead.

**Comment by skywolf829** [comment]
Oops duplicate of #20
